### DISCUSSION TITLE
DistillationDilaDBLoss question

### DISCUSSION THREAD

**Discussion by csj3684** [PaddlePaddle/PaddleOCR#14021] [General]
Hi, thanks for your great works. 

I did OCR detection training with 'ch_PP-OCRv4_det_cml.yml' on my custom dataset.

And I just found there is a something wrong with 'DistillationDilaDBLoss'

It inherits 'DBLoss' and it should send some parameters but in the definition in DistillationDilaDBLoss.__init__, it does not.

Any reason for this ?

Thank you.

class DBLoss(nn.Layer):
    """
    Differentiable Binarization (DB) Loss Function
    args:
        param (dict): the super paramter for DB Loss
    """

    def __init__(
        self,
        balance_loss=True,
        main_loss_type="DiceLoss",
        alpha=5,
        beta=10,
        ohem_ratio=3,
        eps=1e-6,
        **kwargs,
    ):
        super(DBLoss, self).__init__()
        self.alpha = alpha
        self.beta = beta
        self.dice_loss = DiceLoss(eps=eps)
        self.l1_loss = MaskL1Loss(eps=eps)
        self.bce_loss = BalanceLoss(
            balance_loss=balance_loss,
            main_loss_type=main_loss_type,
            negative_ratio=ohem_ratio,
        )

    def forward(self, predicts, labels):
        ...

class DistillationDBLoss(DBLoss):
    def __init__(
        self,
        model_name_list=[],
        balance_loss=True,
        main_loss_type="DiceLoss",
        alpha=5,
        beta=10,
        ohem_ratio=3,
        eps=1e-6,
        name="db",
        **kwargs,
    ):
        **super().__init__()**
        self.model_name_list = model_name_list
        self.name = name
        self.key = None

    def forward(self, predicts, batch):
        loss_dict = {}
        for idx, model_name in enumerate(self.model_name_list):
            out = predicts[model_name]
            if self.key is not None:
                out = out[self.key]
            **loss = super().forward(out, batch)**

            if isinstance(loss, dict):
                for key in loss.keys():
                    if key == "loss":
                        continue
                    name = "{}_{}_{}".format(self.name, model_name, key)
                    loss_dict[name] = loss[key]
            else:
                loss_dict["{}_{}".format(self.name, model_name)] = loss

        loss_dict = _sum_loss(loss_dict)
        return loss_dict
